tests/std/tests/P1135R6_barrier/test.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <atomic>
+#include <barrier>
+#include <cassert>
+#include <thread>
+#include <utility>
+
+void test() {
+    std::barrier b(2);
+
+    std::atomic<int> c{0};
+
+    std::thread t1([&] {
+        for (int i = 0; i < 5; ++i) {
+            auto token = b.arrive();

```

Comment:
I think tests which do more 'adventerous' things with `token` are missing; at least moving it elsewhere and then waiting with the moved one should be necessary.

Reply:
Also missing a test where we are waiting with a previous phase's token which should return immediately.